src/airflow/airqo_etl_utils/airnow_utils.py

```diff
@@ -65,8 +64,15 @@ def extract_bam_data(start_date_time: str, end_date_time: str) -> pd.DataFrame:
     def process_bam_data(data: pd.DataFrame) -> pd.DataFrame:
 
         air_now_data = []
+
+        devices = AirQoApi().get_tenant_devices(tenant=Tenant.US_EMBASSY)
         for _, row in data.iterrows():
             try:
+                device_id = row["FullAQSCode"]
+                device_details = list(
+                    filter(lambda y: str(device_id) in y["device_codes"], devices)
+                )[0]
```

Comment:
Using device code to filter corresponding device data.

Reply:
Brilliant! Thanks!